src/lib.rs

```diff
@@ -13,6 +14,17 @@ mod util;
 
 pg_module_magic!();
 
+pub fn build_pg_list_of_cstrings<'a, I>(parts: I) -> PgList<pg_sys::Value>
```

Comment:
I'm still not quite happy with this. The way we're using it right now is okay, but it's kind of a time bomb.

The following example will compile, but should not:

```rust
    let cstr1 = CString::new("Hello").unwrap();
    let cstr2 = CString::new("World").unwrap();

    let pg_list = crate::build_pg_list_of_cstrings([cstr1.as_bytes(), cstr2.as_bytes()]);

    // This is totally unsafe, `pg_list` is still holding a reference to this.
    drop(cstr1);
```

Ideally we would solve this with lifetimes, but I think that would require reworking (or wrapping) `PgList`. That would be too much work (for now).

As a short-term solution, I would suggest that we mark this function as unsafe, and specify in the function's documentation that the _caller_ must ensure that the parameters which are passed into this function live for longer than the returned `PgList`.

Reply:
How about this:
```rust
pub fn with_pg_list_of_cstrings<'a, I, R, F>(parts: I, f: F) -> R
where
    I: IntoIterator<Item = &'a [u8]>,
    F: Fn(&'a PgList<pg_sys::Value>) -> R,
{
    let mut list = PgList::new();
    for p in parts {
        list.push(unsafe { pg_sys::makeString(p.as_ptr() as _) });
    }
    f(&list)
}
```

Reply:
Usage example:

```rust
            let jsonb_contains_op_oid =
                with_pg_list_of_cstrings(CONTAINS_OP_PATH, |jsonb_contains_fully_qualified_name| {
                    LookupOperName(
                        std::ptr::null_mut(),
                        jsonb_contains_fully_qualified_name.as_ptr(),
                        pg_sys::JSONBOID,
                        pg_sys::JSONBOID,
                        false, // Raises an error if the operator is not found
                        -1,
                    )
                });
```

Reply:
Nevermind, `PgList`'s API is just unsound:

```rust
let mut lst = PgList::<pg_sys::Value>::new();
with_pg_list_of_cstrings(CONTAINS_OP_PATH, |foo| {
    lst.push(foo.pop().unwrap());
});
```